tests/unit/drivers/test_recursive_traversal_tree.py

```diff
@@ -0,0 +1,254 @@
+import os
+import pytest
+
+from jina.proto import jina_pb2
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+from jina.drivers.querylang.slice import SliceQL
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+DOCUMENTS_PER_LEVEL = 2
+
+
+def build_docs():
+    """ Builds up a complete chunk-match structure, with a depth of 2 in both directions recursively. """
+    docs = []
+    for base_id in range(DOCUMENTS_PER_LEVEL):
+        d = jina_pb2.Document()
+        d.granularity = 0
+        d.adjacency = 0
+        d.id = base_id
+        docs.append(d)
+        iterate_build(d, 0, 2, 0, 2)
+    return docs
+
+
+def iterate_build(d, current_granularity, max_granularity, current_adjacency, max_adjacency):
+    if current_granularity < max_granularity:
+        for i in range(DOCUMENTS_PER_LEVEL):
+            dc = d.chunks.add()
+            dc.granularity = current_granularity + 1
+            dc.adjacency = current_adjacency
+            dc.id = i
+            iterate_build(dc, dc.granularity, max_granularity, dc.adjacency, max_adjacency)
+    if current_adjacency < max_adjacency:
+        for i in range(DOCUMENTS_PER_LEVEL):
+            dc = d.matches.add()
+            dc.granularity = current_granularity
+            dc.adjacency = current_adjacency + 1
+            dc.id = i
+            iterate_build(dc, dc.granularity, max_granularity, dc.adjacency, max_adjacency)
+
+
+@pytest.mark.skip('this is a test for the proposed new flow design')
```

Comment:
pytest.mark.skip requries `reason` as kwargs, so u need to put `reason='this is ...`